crates/credential/cargo-credential-wincred/src/main.rs

```diff
@@ -0,0 +1,93 @@
+//! Cargo registry windows credential process.
+
+use cargo_credential::{Credential, Error};
+use std::ffi::OsStr;
+use std::os::windows::ffi::OsStrExt;
+use winapi::shared::minwindef::{DWORD, FILETIME, LPBYTE, TRUE};
+use winapi::shared::winerror;
+use winapi::um::wincred;
+use winapi::um::winnt::LPWSTR;
+
+struct WindowsCredential;
+
+fn wstr(s: &str) -> Vec<u16> {
+    OsStr::new(s).encode_wide().chain(Some(0)).collect()
```

Comment:
This may want to return an error as well if `s` contains a 0, because otherwise I think it'll get silently truncated